week1/assignment3-2/ex4.cpp

```diff
@@ -0,0 +1,244 @@
+    double amount{};
+    std::cin >> amount;
+    std::cout << "\n\nAccount balance after withdrawal is " 
+        << (customer.accounts[acc_num].balance -= amount) << " €\n";
+    return customer;
+    
+}
+
+int get_account_number() {
+    std::cout << "Enter the account number: ";
+    int acc_num{};
+    std::cin >> acc_num;
+    return acc_num;
+}
+
+void create_new_account(Customer& customer) {
+    std::cout << "Enter the amount you want to deposit: ";
+    int amount{};
+    std::cin >> amount;
+    // global static int account_number increases by 1
+    customer.accounts[++account_number].balance = {amount};  
+    std::cout << "A new account created with number " << account_number 
+        << " and with balance " << customer.accounts[account_number].balance << " €\n";
+}
+
+void create_new_customer() {
+    // Just for checking if there is "some rubbish in the cin" to avoid program act weirdly
+    if (!std::cin) {   
+        std::cin.clear();
+    }
+    std::cin.ignore(std::numeric_limits<std::streamsize>::max(),'\n');
+    
+    std::cout << "Name: ";
+    std::string n{};
+    std::getline(std::cin, n);
+
+    std::cout << "Address: ";
+    std::string a{};
+    std::getline(std::cin, a);
+
+    std::cout << "Phone number: ";
+    std::string p{};
+    std::getline(std::cin, p);
+
+    std::cout << "Balance: ";
+    int b{};
+    std::cin >> b;
+
+    // create customer
+    Customer customer = {.name = n, .address = a, .phone = p, {}};
+
+    // create account and connect it to the customer
+    Account acc = {b};
+    customer.accounts[++account_number] = acc;
+
+    // Connect customer to the bank db
+    bank[++customer_number] = customer;
+
+    std::cout << "\nA new customer '" << customer.name << "' created" 
+              << " with customer number " << customer_number << "\n";
+    std::cout << "And a new account created with account number " << account_number 
+              << " and with balance " << b << "\n";
+    
+    // Save data to the csv-file
+    save_customer_data("customer_data.csv", bank[customer_number]);
+}
+
+void save_customer_data(const std::string& filename, Customer &current_customer) {
+    std::ofstream customer_data_file(filename);  // Create and open a data file
+    if (customer_data_file.is_open()) {
+        customer_data_file << "1.Customer number, 2.Name, 3.Address, 4.Phone number, "
+                              "5. Account number, 6.Balance\n";
+        // Loop through the maps and write customer data to the file
+        for (const auto& [customer_number, current_customer] : bank) {
+            for (const auto& [account_number, account] : current_customer.accounts) {
+                customer_data_file << "1." << customer_number << ", 2." << current_customer.name 
+                << ", 3." << current_customer.address << ", 4." << current_customer.phone << ", 5." 
+                << account_number << ", 6." << account.balance << "\n";
+            }
+        }
```

Comment:
This is a bit hard to read, so to make it more readable you could add line breaks here. Although it is quite obvious what's happening here since it's well commented, so it's not necessary.
```suggestion
        for (const auto& [customer_number, current_customer] : bank) {
            for (const auto& [account_number, account] : current_customer.accounts) {
                customer_data_file
                        << "1." << customer_number
                        << ", 2." << current_customer.name 
                        << ", 3." << current_customer.address
                        << ", 4." << current_customer.phone
                        << ", 5." << account_number
                        << ", 6." << account.balance << "\n";
            }
        }
```